app/batid/services/guess_bdg_new.py

```diff
@@ -401,3 +433,121 @@ def _geocode_name_and_point(name: str, lat: float, lng: float) -> Optional[Point
             return Point(lng, lat, srid=4326)
         else:
             return
+
+
+class PartialRoofHandler(AbstractHandler):
+    """
+    This handler is used to match a building based on a roof section.
+    """
+
+    _name = "partial_roof"
+
+    def __init__(self, isolated_section_max_distance=1, min_second_bdg_distance=6):
+        self.isolated_section_max_distance = isolated_section_max_distance
+        self.min_second_bdg_distance = min_second_bdg_distance
+
+    def _guess_batch(self, guesses: dict) -> dict:
+        tasks = []
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            for guess in guesses.values():
+                future = executor.submit(self._guess_one, guess)
+                future.add_done_callback(lambda future: connections.close_all())
+                tasks.append(future)
+
+            for future in concurrent.futures.as_completed(tasks):
+                guess = future.result()
+                guesses[guess["input"]["ext_id"]] = guess
+
+        return guesses
+
+    def _guess_one(self, guess: dict) -> dict:
+        roof_geojson = guess["input"].get("polygon", None)
+
+        if not roof_geojson:
+            return guess
+
+        roof_poly = GEOSGeometry(json.dumps(roof_geojson))
+
+        # Get the two closest buildings
+        closest_bdgs = get_closest_from_poly(roof_poly, 35)[:20]
```

Comment:
J'ai l'impression que c'est une typo

```suggestion
        closest_bdgs = get_closest_from_poly(roof_poly, 35)[:2]
```

Reply:
Ce n'est pas une typo. On a bien besoin de récupérer plus de bâtiments (je dois avouer ne plus me souvenir pourquoi car c'est un peu lointain, je me souviens que ça a à faire avec le fait que plusieurs bâtiments peuvent avoir une distance de zéro).